### ISSUE TITLE
cloud builder for dotnet 2.0

### ISSUE THREAD

**Issue by sursani** [GoogleCloudPlatform/cloud-builders#145]
Hello,

I am trying to utilize the cloudbuilder.yaml template at the location here alongside the GCP container registry -> build trigger feature:
[https://github.com/GoogleCloudPlatform/cloud-builders/tree/master/csharp/dotnet/examples/TestApp-2.0](url)

However, when I kick off the build, I get this error:

> MSBUILD : error MSB1003: Specify a project or solution file. The current working directory does not contain a project or solution file.

It is happening on this step:
```
Arguments	
restore --packages /packages
```
This is line 7 in the cloudbuild.yaml file located in your github project above.

The only difference I can think of is that my Git repo has the webapi project which I am attempting to build one folder level deep. So in GCP build trigger definition it looks like this:

> myfoldername/cloudbuild.yaml

Also note, my cloudbuild.yaml file is sitting next to the csproj build which it claims it can't find in the error above. I am guessing that even though GCP is pointing to my cloudbuild.yaml file one folder in, the working directory is still the root folder of my repo where the csproj is not present.

Any thoughts?

**Comment by ivannaranjo** [comment]
Hello,
It all really depends on how you are invoking the gcloud command to submit the build. From your description it seems that you are doing the following:
```bash
gcloud container builds submit --config myfoldername/cloudbuild.yaml .
```

In this situation the "current directory" when the cloudbuild.yaml is executed is going to be one above where the .csproj resides, which will explain why the `dotnet` command cannot find it. To fix this you should also specify "myfoldername" as the directory to build. Or more simply:
```bash
cd myfoldername
gcloud container builds submit --config cloudbuild.yaml .
```

This is also the best way to build the examples.

Is this correct? If not, can you share your `gcloud` command as well?

**Comment by sursani** [comment]
Hi Ivan,
Thanks for your feedback. I was actually trying this using the GCP "Build Triggers" inside the Container Registry section. Upon check in to my github repo's master branch, I am wanting my build trigger to kick off.

See below:
![image](https://user-images.githubusercontent.com/527538/31723968-ff3f1aa6-b3e5-11e7-9ed4-672b7003bd3e.png)

![image](https://user-images.githubusercontent.com/527538/31724222-c01d2aba-b3e6-11e7-83fb-11535c2e7545.png)

So as shown in the screenshot, I am pointing to the cloudbuild.yaml in a subfolder but the Git repo is one level above. How would I accomplish a CD myfoldername in the cloudbuild.yaml (I copied yours from my first post)?

![image](https://user-images.githubusercontent.com/527538/31724430-43eeb9b2-b3e7-11e7-8107-e6938d4cd5a7.png)

On the flip side, I created a brand new project where my github repo has the csproj and cloudbuild.yaml at the top level and the build trigger works beautifully.
